### DISCUSSION TITLE
Install Error "There was a problem accessing the file system on the server" when adding Modern Search V4 to SharePoint site

### DISCUSSION THREAD

**Discussion by ragun101** [microsoft-search/pnp-modern-search#3347] [Q&A]
**Version used**
Ex: 4.9.3.0

**Describe the bug**
I recently updated the SharePoint Modern Search v4 App to the newest version. When I add the app to a site and navigate to the App Details page I see an install error stating the following:

`There was a problem accessing the file system on the server.
Details: Deployment failed in host web https://<mytenantname>.sharepoint.com/sites/<mysitename> for app PnP Modern Search - Search Web Parts - v4/59903278-dd5d-4e9e-bef6-562aae716b8b. 
Microsoft.SharePoint.SPException: An error has occurred activating solution 59903278-dd5d-4e9e-bef6-562aae716b8b. 
Failed to add file Template\Features\544c1372-77ae-40a0-9095-805ae2a0d5e9\544c1372-77ae-40a0-9095-805ae2a0d5e9\WebPart_544c1372-77ae-40a0-9095-805ae2a0d5e9.xml. 
	at Microsoft.SharePoint.Administration.SPUserCodeSolutionPackage.AddToSolutionsTable(SPSite site, SPUserSolution solution, String solutionValidatorsHash, Int32 solutionGalleryItemId, Dictionary'2 solutionFiles) 
	at Microsoft.SharePoint.Packaging.SPTargetWebDeploymentGroup.GenerateAndActivateUserSolutionFromPackage(SPWeb web, SPAppPackage appPackage, Guid productId, String solutionName, SPSolutionLevel solutionLevel) 
	at Microsoft.SharePoint.Packaging.SPTargetWebDeploymentGroup.InstallOrUpgrade(SPSite site, Nullable'1& solutionId, Boolean& swapNeeded)`

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Go to "Site contents" 
2. Click on "add an app"
3. Click on "App details" > "PnP Modern Search - Search Web Parts - v4"
4. Click on "Request it"
5. Go back to "Site contents"
6. Click on "..." on "PnP Modern Search - Search Web Parts - v4" > "App Details"
7. See error

**Expected behavior**
The app is expected to be added correctly and for the webparts to be selectable when editing a page. 

**Desktop (please complete the following information):**
 - Browser Microsoft Edge
 - Executing as SharePoint Administrator
 - Previous Version 4.8.0.0 worked perfectly fine

**Additional context**
The error is not displayed during the install. The app appears in the site content as expected. I could not select the WebParts when editing a site and therefore checked if there were any errors during the installation. The four PnP Search features are visible in the ManageFeatures.aspx and are also active.

**Comment by wobba** [comment]
Updating an app can take some time for the CDN and all caching to be resolved. Always to it over the weekend. If you fail to update, then delete the app, and re-add,

**Reply by ragun101** [reply]
Thank you for your reply. If I remove the app and re-add it, will the pages that already use Modern Search V4 break or will they work fine as soon as the app is re-added? And would you recommend to remove older versions all together? Thanks!

**Reply by wobba** [reply]
It's perfectly safe to remove an app and re-add. All pages will continue to work :) They might break in the time it takes for the new app to be added, but that's about it. And you can only keep one version of the web part at a time (globally). v3 and v4 are individual web parts and can co-exist. But only one version for each of them.
